### ISSUE TITLE
Filesystem Provider, UNC Paths & Double Backslash

### ISSUE THREAD

**Issue by FriedrichWeinmann** [PowerShell/PowerShell#13633]
## Steps to reproduce

Run Remove-Item with a UNC path that contains a doubled backslash before the last item and including `-WhatIf`.
After confirming that you want to delete multiple items even though you didn't specify recurse, check the path it would have deleted.

```powershell
Remove-Item \\localhost\C$\temp\\ADMF -WhatIf
```

## Expected behavior

```none
What if: Performing the operation "Remove Directory" on target "\\localhost\C$\temp\ADMF"
```

## Actual behavior

```none
What if: Performing the operation "Remove Directory" on target "\\localhost\C$\temp\"
```

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.1.0-preview.6
PSEdition                      Core
GitCommitId                    7.1.0-preview.6
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by rjmholt** [comment]
We should consider fixing this for RC.1

**Comment by TravisEz13** [comment]
Calling this an enhancement while we check if this is a regression.

**Comment by jjolidon** [comment]
Just to confirm it is still biting on 7.2.1.

**Comment by ajansveld** [comment]
Why is this considered an "enhancement"? We just had our first incident because of this destructive bug, I feel we should be much louder about this.

Is this being looked at for 7.3?

**Comment by Tadas** [comment]
Looks like this has been fixed with #15873

**Comment by ajansveld** [comment]
What is the threshold for backporting the fix to 7.2 (LTS-current), considering it is supported for another 2 years?

**Comment by iSazonov** [comment]
@ajansveld Only security fixes are backported. Also regressions from previous version.
